### PR TITLE
src/format: use u32 and std::process::id not libc::getpid

### LINKED ISSUE #46: Formatter*::pid takes i32, but std::process::id returns u32
All,

In order to run properly, I must do the following:

```
    /* get our process id (syslog takes i32, but we get u32) */
    let p = process::id();
    let pid = if p > std::i32::MAX as u32 {
        0
    } else {
        p as i32
    };

    /* format for our syslog messages */
    let formatter = Formatter3164 {
        facility: Facility::LOG_USER,
        hostname: None,
        process: "myrustprocess".into(),
        pid: pid,
    };
```

If we could change the type of `pid` to be `u32`, then we wouldn't have to jump through the overflow check...

I've checked and both `Formatter*` structs are `i32` for `pid`

Thanks!

Jason

### PR DESCRIPTION
**Pull request by arnottcr**
Since 0.3.2, libc::getpid is used to get the process id and store the
provided id in Formatters. This has the marked side effect of being
unsafe and adding an unnecessary crate dependency. As such, this change
moves to using std::process::id, which likely was not stable at the
time, for better stdlib integration.

Unfortunately, this is a breaking change, since the current i32s are
part of the external interface, but could be implemented as compatible
if the extant symbols were simply deprecated and new ones added.

Source has also been formatted and linted via rustfmt and clippy, and
dependencies upgraded. Time 0.2 was held back since it incompatible
with 0.1.

Fix #46

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,10 +10,10 @@ documentation = "https://docs.rs/syslog"
 keywords = ["syslog", "logs", "logging"]
 
 [dependencies]
-libc        = "^0.2"
-time        = "^0.1"
-log         = { version =  "^0.4", features = [ "std" ] }
-error-chain = { version = "^0.12", default-features = false }
+libc        = "0.2.67"
```
Comment (daboross) [inline Cargo.toml]: I don't think anything else in this library uses `libc`, so I think it should be possible to just drop it?

Comment (arnottcr) [reply Cargo.toml]: Unfortunately all os consts only live in `libc` not `std` and we leverage [`libc::EPROTOTYPE`](https://github.com/Geal/rust-syslog/blob/92a087169743699c489d6a5fd5c857017862b59a/src/lib.rs#L263).

**Comment by arnottcr** [comment]
> It _was_ pretty hard to tell what's changed though? ... [I]t's usually good etiquette to make formatting overhauls and substantive changes completely separate, so they can be reviewed separately.

Sorry, my editor runs rustfmt by default and the clippy changes looked pretty minor. I have extracted all of both into #50.

**Comment by Geal** [comment]
it is a good idea. There's still one use of libc that I'll try to remove: https://github.com/Geal/rust-syslog/blob/020ccf1a8f1b05b51bd136937250f9e67f522af6/src/lib.rs#L267

**Pull request merged**